### DISCUSSION TITLE
Trigger event on change of Order's Paid Status

### DISCUSSION THREAD

**Discussion by nfourtythree** [craftcms/commerce#2418] [Ideas]
When an order's paid status changes trigger an event.

An implementation strategy is required for this due to the number of variables as to why the paid status might change.

As just one example an order might go from `paid` to `unpaid` for the following reasons:

- Order was fully or partially refunded
- A transaction was deleted
- Order was edited in the control panel
- Order was edited programmatically

**Comment by mikejpeters** [comment]
Possibly related - is there a recommended way to mark an order as cancelled? We are using Commerce for selling courses, so to list courses a user "owns" we want to get a list of orders that weren't refunded. Using the `paid`/`unpaid` status kind of works except in special cases like where an order was partially refunded or was manually entered into the control panel. Do we need to use [custom order statuses](https://craftcms.com/docs/commerce/4.x/custom-order-statuses.html) for this?
